code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/FilmstripCarouselTemplateNotificationBuilder.kt

```diff
@@ -0,0 +1,169 @@
+            newIndices = CarouselTemplateHelpers.calculateNewIndices(
+                centerImageIndex,
+                downloadedImageUris.size,
+                pushTemplate.intentAction
+            )
+            pushTemplate.centerImageIndex = newIndices[1]
+        } else {
+            newIndices = listOf(centerImageIndex - 1, centerImageIndex, centerImageIndex + 1)
+        }
+
+        // set the carousel images in the filmstrip carousel
+        populateFilmstripCarouselImages(
+            context,
+            downloadedImages as List<Bitmap?>,
+            imageCaptions as List<String?>,
+            imageClickActions as List<String?>,
+            newIndices,
+            pushTemplate,
+            trackerActivityName,
+            expandedLayout
+        )
+
+        val titleText = pushTemplate.title
+        val smallBodyText = pushTemplate.body
+        val expandedBodyText = pushTemplate.expandedBodyText
+        smallLayout.setTextViewText(R.id.notification_title, titleText)
+        smallLayout.setTextViewText(R.id.notification_body, smallBodyText)
+        expandedLayout.setTextViewText(R.id.notification_title, titleText)
+        expandedLayout.setTextViewText(R.id.notification_body_expanded, expandedBodyText)
+
+        // Create the notification channel if needed
+        channelIdToUse = createChannelAndGetChannelID(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance()
+        )
+
+        // handle left and right navigation buttons
+        val clickIntent = createClickIntent(
+            context,
+            pushTemplate,
+            PushTemplateConstants.IntentActions.FILMSTRIP_LEFT_CLICKED,
+            broadcastReceiverName,
+            trackerActivityName,
+            downloadedImageUris as List<String?>,
+            imageCaptions,
+            imageClickActions
+        )
+        val pendingIntentLeftButton = PendingIntent.getBroadcast(
+            context,
+            0,
+            clickIntent,
+            PendingIntent.FLAG_MUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        )
+        clickIntent.setAction(PushTemplateConstants.IntentActions.FILMSTRIP_RIGHT_CLICKED)
+        val pendingIntentRightButton = PendingIntent.getBroadcast(
+            context,
+            0,
+            clickIntent,
+            PendingIntent.FLAG_MUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        )
+
+        // set onclick intents for the skip left and skip right buttons
+        expandedLayout.setOnClickPendingIntent(R.id.leftImageButton, pendingIntentLeftButton)
+        expandedLayout.setOnClickPendingIntent(R.id.rightImageButton, pendingIntentRightButton)
+
+        // create the notification builder with the common settings applied
+        return super.construct(
+            context,
+            pushTemplate,
+            trackerActivityName,
+            smallLayout,
+            expandedLayout
+        )
```

Comment:
After transforming `AEPPushTemplateBuilder` to an object can you apply the default construction properties first and then apply the specific properties. This will prevent any custom properties set in this method from being modified again due to this call.